tests/client.rs

```diff
@@ -3,6 +3,18 @@ use std::net::SocketAddr;
 use tokio::net::TcpListener;
 use tokio::task::JoinHandle;
 
+/// A basic PING PONG test.
+#[tokio::test]
+async fn ping_pong() {
+    let (addr, _) = start_server().await;
+    let mut client = client::connect(addr).await.unwrap();
+    let pong = client.ping(None).await.unwrap();
+    assert_eq!(b"PONG", &pong[..]);
+
+    let pong = client.ping(Some("你好世界".to_string())).await.unwrap();
```

Comment:
nit: I would go with keeping each test case test one thing and split this test case into 2: one for an empty message and another for a non-empty message.